Assets/Scripts/Actors/BuddyStats.cs

```diff
@@ -13,27 +27,80 @@ public bool isAlive
 		get { return _isAlive; }
 	}
 
-	int currentStats = 0;
 	ParticleSystem _particles;
 
+	uint ID = 0;
+
 	void Awake()
 	{
+		ID = GetID(); // Grab the current unique ID
+		name = "Buddy " + GetRandomName( ID );
+		currResourceTimer = decreaseResourcesTime;
+		apples = startingApples;
 		_particles = GetComponentInChildren<ParticleSystem>();
+		SetGod(owner); // For owners set in the inspector
+	}
+
+	public void SetGod( GodTag _godTag )
+	{
+		if(_godTag)
+		{
+			owner = _godTag;
+			_actorStats = _godTag.gameObject.GetComponent<PlayerActorStats>();
+		}
+	}
+
+	static string[] names = { "Longnose", "Jojo", "JillyJane", "Sunshine", "Moosejaw"};
+	static uint uniqueID = 1;
+
+	static uint GetID()
+	{
+		return uniqueID++;
+	}
+
+	static string GetRandomName( uint ID )
+	{
+		int randIndex = Random.Range( 0, names.Length );
+
+		return names[randIndex] + ID;
 	}
 
 	void Update()
 	{
-		if ( Input.GetKeyDown( KeyCode.K ) )
+		// Replace this with an invoke
+		currResourceTimer -= Time.deltaTime;
+
+		if ( currResourceTimer < 0.0f && _isAlive )
+		{
+			// decrease resources
+			apples--;
```

Comment:
I don't think this timer stuff matters much. There won't be that many buddies on screen at a time and if it does become an issue we can rework it into an invoke function instead of the update with a timer.